app/src/main/java/com/duckduckgo/app/browser/BrowserTabFragment.kt

```diff
@@ -1087,10 +1087,21 @@ class BrowserTabFragment : Fragment(), FindListener, CoroutineScope, DaxDialogLi
         }
     }
 
+    private fun closeAndReturnToSourceIfBlankTab() {
+        if (viewModel.url == null) {
+            launch {
+                dismissDownloadFragment()
+                destroyWebView()
```

Comment:
Makes complete sense. I think I threw this in trying to squash the memory leak. I'll take it and the `dismessDownloadFragment` out so that the fragment lifecycle hands this instead.